### PR TITLE
CSS-IN-JS POC

### PR DESCRIPTION
**Pull request by JoshuaRogan**
## Reasons to Switch
* Clients need not worry about webpack/scss/global styles!!!!
* CSS in JS abstracts style to the component level.
* Doesn't require any web pack config for clients
  *  no build step, as a result—no dependency to build tools at all.
* Proper isolation preventing global side effects from greedy selectors (e.g. p)
  * No more global styles to worry about...everything is in the component
* Scalable 
* **Opens a path to react native with one codebase**
* Power of theming
* Runtime prefixes
* JSS styles size is up to 50% smaller
* Coupling of css with js allows using properties to compute styles 
  *  easily share constants and functions between JS and CSS\
* All CSS features included.
* Easy SSR

## Bundle Size
jss & react-jss adds ~9K gzipped which is nothing compared to the cost of removing all of the scss files and not having to worry about serving them. 

## Links to Share
* [Syntax Examples](http://cssinjs.org/json-api?v=v9.8.7#media-queries)
* [Creating Design System with css-in-js](https://medium.freecodecamp.org/css-in-javascript-the-future-of-component-based-styling-70b161a79a32)
* [Pros Cons](https://objectpartners.com/2017/11/03/css-in-js-benefits-drawbacks-and-tooling/)

screen_shot_2018-07-28_at_4 31 58_pm

## Components to Convert
It takes about 10-20 minutes to convert each at this point.

 - [X] Button
 - [ ] ButtonGroup
 - [ ] FloatingButton
 - [ ] FloatingButtonGroup
 - [X] Spinner
 - [ ] Input
 - [X] WDSIcon (New)
 - [ ] Fieldset
 - [X] BannerNotification
 - [ ] BannerNotifications
 - [ ] VideoPlayIcon
 - [ ] ContentWell
 - [ ] FandomContentWell
 - [ ] FandomBackgroundImage
 - [ ] Vignette
 - [ ] ExpandableText

### CODE REVIEW

File: src/components/BannerNotification/index.js [hunk 1]
```diff
@@ -1,48 +1,82 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import './styles.scss';
```
Comment (JoshuaRogan) [inline src/components/BannerNotification/index.js]: 🙌

**Comment by JoshuaRogan** [comment]
@vforge I'm curious if you would be willing to discuss this again. The FC will be implementing Theming soon and we want to have the capacity to theme stuff in the RDS. Maybe css-in-js is not the best option for an application but I think it's really great for a theme that needs theming  https://wikia-inc.atlassian.net/wiki/spaces/CAKE/blog/2018/10/25/445939726/CSS-IN-JS+Exploration

**Comment by vforge** [comment]
@JoshuaRogan As I also did some experiments with those I'll schedule a meeting with you this week - let's talk solutions.

**Comment by vforge** [comment]
Please update and reopen PR if needed.

**Pull request closed without merging**